chcemvediet/apps/inforequests/admin.py

```diff
@@ -315,3 +317,58 @@ def get_queryset(self, request):
         queryset = queryset.select_related(u'branch')
         queryset = queryset.select_related(u'email')
         return queryset
+
+    def get_actions(self, request):
+        actions = super(ActionAdmin, self).get_actions(request)
+        if u'delete_selected' in actions:
+            del actions[u'delete_selected']
+        return actions
+
+    def has_delete_permission(self, request, obj=None):
+        if obj is None:
+            return True
+        if obj.type in [Action.TYPES.REQUEST, Action.TYPES.ADVANCED_REQUEST]:
+            return False
+        if len(obj.branch.actions) > 1:
+            return True
+        return False
+
+    def render_delete_form(self, request, context):
+        using = router.db_for_write(self.model)
+        collector = admin.utils.NestedObjects(using)
+        collector.collect([context[u'object']])
+        to_delete = collector.nested()
+        for object in traverse(to_delete):
+            if isinstance(object, Action) and object.email:
+                for inforequestemail in object.email.inforequestemail_set.filter(
+                        type=InforequestEmail.TYPES.APPLICANT_ACTION):
+                    context[u'deleted_objects'].append(
+                        format_inforequestemail(inforequestemail, self.admin_site))
+        return super(ActionAdmin, self).render_delete_form(request, context)
+
+    def delete_model(self, request, obj):
+        using = router.db_for_write(self.model)
```

Comment:
@martinmacko47 mam si spravit tiez pomocnu funkciu, kedze sa mi to opakuje v `render_delete_form` a `delete_model`?

Reply:
@viliambalaz jj, to by bolo fajn. Teda nie ani tak funkciu ako metodu. Nieco ako "collect nested inforequestemail" alebo taky nejaky nazov. Ktora vrati dve polia: zoznam obligee action inforequestemailov a zoznam applicant action inforequestemailov.

Reply:
@martinmacko47 . Nerozumiem, naco mi bude sluzit parameter 2 polia obligee/applicant inforequestemailov? Prave tieto objekty chcem zistit. Skor by to mala byt navratova hodnota metody.
Taktiez mi napadlo namiesto pola 
```
object.email.inforequestemail_set.filter(type=InforequestEmail.TYPES.APPLICANT_ACTION)
```
pouzit QuerySet, nieco ako
```
outbound = InforequestEmail.objects.filter(inforequest__branch__action__in=actions, email__in=emails, type=InforequestEmail.TYPES.APPLICANT_ACTION).distinct()
```
pricom `actions` su instancie vratene z `NestedObjects`.

Reply:
@viliambalaz sorry, vypadlo mi tam slovo "a **vrati** dve polia"

S tym QuerySet namiesto pola je dobry napad. Akurat by zrejme malo stacit:

```
outbound = InforequestEmail.objects.filter(inforequest=action.branch.inforequest, email__in=emails, type=...)
```

Kedze `emails` si zobral z nested actions, tak netrena explicitne znovu filtrovat, ze patria akciam. Len treba zabespecit, ze patria spravnej infoziadosti. Vdaka comu nebudu potrebne ani ziadne joiny a teda ani `.distinct()`. Ci som nieco prehliadol?

A teda vdaka tomu, ze `outbound` a `inbound` budu QuerySety, tak sa bude dat napisat rovno `outbound.delete()` a `inbound.update(...)`. Cim teda vlastne vsetko spravime iba v dvoch query.

Reply:
:ok: